### PR TITLE
Implementing CODAP export

### PR DESCRIPTION
**Pull request by IshaGupta18**
@jywarren @namangupta01 as you know I have been in conversation with @jsondoe from CODAP who has been helping me out a lot, so I began implementing the feature, after a lot of exploration and discussion on the plugin API. I am facing just a few issues in importing one of the files `iframe-phone.js`, apart from which, I was able to test the functioning in an example tester.

Here's how to works in the example and this is kind of what we will be implementing.

codap

And this is what is ready till now!

codapsdg

I have mailed @jsondoe about the issue and once we fix it, the feature will be ready! I once again thank him for all this help!

### CODE REVIEW

File: dist/transpiled_code/CsvParser.js [hunk 1]
```diff
@@ -282,6 +314,20 @@ function () {
       }
 
       return headers_sheet;
+    } //determine a JSON for headers for CODAP
+
+  }, {
```
Comment (IgorWilbert) [inline dist/transpiled_code/CsvParser.js]: The code formatting looks great, but is there any way of splitting it a bit? With so many parentheses, it is a bit hard to know which are the pairs that match :)

File: dist/transpiled_code/indexsheet.js [hunk 2]
```diff
@@ -12,64 +18,72 @@ window.addEventListener('load', function () {
...
+
+  callback(oAuth2Client); // });
+}
+
+function listMajors(auth) {
+  var sheets = google.sheets({
+    version: 'v4',
+    auth: auth
+  }); //     // sheets.spreadsheets.values.get({
+  //     //     spreadsheetId: '1BxiMVs0XRA5nFMdKvBdBZjgmUUqptlbs74OgvE2upms',
```
Comment (IgorWilbert) [inline dist/transpiled_code/indexsheet.js]: I suppose we'll get rid of some commented code along the way?

File: src/iframe-phone.js [hunk 3]
```diff
@@ -0,0 +1,464 @@
+!function(e){"object"==typeof exports?module.exports=e():"function"==typeof define&&define.amd?define(e):"undefined"!=typeof window?window.iframePhone=e():"undefined"!=typeof global?global.iframePhone=e():"undefined"!=typeof self&&(self.iframePhone=e())}(function(){var define,module,exports;return (function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);throw new Error("Cannot find module '"+o+"'")}var f=n[o]={exports:{}};t[o][0].call(f.exports,function(e){var n=t[o][1][e];return s(n?n:e)},f,f.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
```
Comment (IgorWilbert) [inline src/iframe-phone.js]: Are you still having trouble with this file?

**Comment by IgorWilbert** [comment]
Awesome job @IshaGupta18 ! I left some comments, I hope they are helpful. Thank you!

**Comment by IshaGupta18** [comment]
Okay so I have resolved the problems, and this is the latest PR, which covers other things as well.

**Comment by IshaGupta18** [comment]
Here's the functioning of CODAP, which is working now!
codapf

@jywarren @namangupta01 @IgorWilbert @gauravano @sagarpreet-chadha let me know if it's good to go!

Thanks a lot @jsandoe for your constant support and help!

**Comment by jywarren** [comment]
Wow this is really exciting. Is there a gh-pages demo I can try out? But go ahead and merge, this is tremendous!

**Comment by IshaGupta18** [comment]
Oh thanks a lot! I will do it!

**Pull request merged**